pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -0,0 +1,264 @@
+
+        Defaults have been set and options have been exposed, but may require configuration
+        based on your level of memory/speed efficiency. We suggest having a look at this tutorial for
+        more information.
+        `https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html`
+
+        Arguments:
+            cpu_offload (Optional [CPUOffload]):
+                CPU offloading config. Currently, only parameter and gradient CPU
+                offload is supported. It can be enabled via passing in
+                ``cpu_offload=CPUOffload(offload_params=True)``. Note that this
+                currently implicitly enables gradient offloading to CPU in order for
+                params and grads to be on same device to work with optimizer. This
+                API is subject to change. Default is ``None`` in which case there
+                will be no offloading.
+            backward_prefetch: (Optional[BackwardPrefetch]):
+                This is an experimental feature that is subject to change in the
+                the near future. It allows users to enable two different backward_prefetch
+                algorithms to help backward communication and computation overlapping.
+                Pros and cons of each algorithm is explained in the class ``BackwardPrefetch``.
+        """
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            cluster_environment=cluster_environment,
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+        self._process_group = None
+        self.num_processes = len(self.parallel_devices) if self.parallel_devices is not None else 0
+        self._has_loaded_state_dict: bool = False
+        self.cpu_offload = cpu_offload
+        self.backward_prefetch = backward_prefetch
+
+    @property
+    def root_device(self) -> torch.device:
+        return self.parallel_devices[self.local_rank]
+
+    @property
+    def process_group(self):
+        if self._process_group is None:
+            # The plugin should have already initilized ddp in setup_environment()
+            self._process_group = _get_default_group()
+        return self._process_group
+
+    @property
+    def setup_optimizers_in_pre_dispatch(self) -> bool:
+        # Setup optimizers after the Fully Sharded Model has been made
+        return True
+
+    def setup_environment(self) -> None:
+        self.setup_distributed()
+        super().setup_environment()
+
+    def setup_distributed(self) -> None:
+        if not self.root_device.type == "cuda":
+            raise MisconfigurationException(
+                "You selected strategy to be `ddp_fully_sharded_native`, but GPU is not available."
+            )
+        reset_seed()
+        # set warning rank
+        rank_zero_only.rank = self.global_rank
+        init_dist_connection(self.cluster_environment, self.torch_distributed_backend)
```

Comment:
torch_distributed_backend is deprecated could you update to https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/strategies/ddp.py#L188-L189?